docs/DeveloperGuide.adoc

```diff
@@ -295,88 +475,155 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|=======================================================================
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is the `ZeroToOne` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
+=== Use Case 1: Adding Exercises to Workout
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User requests for the existing exercises in the workout
+2. System shows a list of the existing exercises
+3. User requests to add an exercise to the workout
+4. System adds the exercise to the workout
 +
 Use case ends.
 
 *Extensions*
 
 [none]
-* 2a. The list is empty.
+* 2a. The requested workout does not exist
+** 2a1. Use case ends
++
+
+* 2b. There are no exercises in the workout
+** 2b1. System shows a message to inform the user
++
+Use case resumes at step 2
+
+* 3a. User request format is invalid
+** 3a1. System shows an error message
++
+Use case resumes at step 3
+
+[discrete]
+=== Use Case 2: Deleting Workouts
+
+*MSS*
+
+1. User requests to list workouts
+2. System shows a list of all workouts
+3. User requests to delete a specific workout
+4. System deletes the workout
 +
 Use case ends.
 
-* 3a. The given index is invalid.
+*Extensions*
+
+[none]
+* 2a. The list is empty
++
+Use case ends
+
+* 3a. The given user index is invalid
+** 3a1. System shows an error message
++
+Use case resumes at step 2
+
+[discrete]
+=== Use Case 3: Find Past Workout by Keyword
+
+*MSS*
+
+1. User requests a log of past workouts
+2. System shows a list of all past logged workouts
+3. User requests to find a past workout by keyword
+4. System shows the past workouts that match the keyword
 +
+Use case ends
+
+*Extensions*
+
 [none]
-** 3a1. AddressBook shows an error message.
+* 2a. The log is empty
 +
-Use case resumes at step 2.
+Use case ends
 
-_{More to be added}_
+* 3a. The keyword does not match any past workouts
+** 3a1. System shows an error message
++
+Use case ends
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+.  Should be able to hold up to 100 workouts a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-_{More to be added}_
+. Should not depend on a remote server, so that a user can exercise in any condition or environment.
 
 [appendix]
 == Glossary
```

Comment:
Would be better if you could add `Workout` and `Session` as well and any other terms you feel are relevant.